src/pages/ScholarshipForm.jsx

```diff
@@ -19,8 +20,19 @@ const useStyles = makeStyles((theme) => ({
 
 function ScholarshipForm() {
   const classes = useStyles();
-  const [submissionAlert, setSubmissionAlert] = useState(false);
-  const [error, setError] = useState(false);
+  const [showAlert, setShowAlert] = useState(false);
+  const [submissionAlert, setSubmissionAlert] = useState(null);
+
+  function showSubmissionAlert(severity, message) {
+    return (
+      <Alert
+        variant="filled"
+        onClose={() => setShowAlert(false)}
+        severity={severity}>
+        {message}
+      </Alert>
+    );
+  }
```

Comment:
A few things:
1. It might be nice to use an `AlertTitle` here ([demo](https://github.com/knative/client/blob/3da89001a2a1d514554a8fa20b35764cc5d2f5c3/pkg/kn/commands/service/configuration_edit_flags.go#L209)) along with a description. The title can be a single word or a short message while the description contain additional details (e.g. the exact error returned)
1. An [action](https://material-ui.com/components/alert/#actions) to view the Scholarship would be nice. You might need to add the `close` icon action as well if `onClose` doesn't play well with `action` 
1. I think the default `variant` is probably fine.
1. I'm not sure how much this function really deduplicates between the two alerts

Reply:
I can probably just set the state directly without the function. 
Regarding the action to show the scholarship, how would I get the id? The `scholarship` object returned in the `.then()` is the scholarship that was submitted.

Reply:
`scholarship.id` should work as `save()` resolves to [`this`](https://github.com/beyondhb1079/s4us/blob/dd57b640aeb6bbff61f1c9ba196816503e8f9b36/src/models/base/FirestoreModel.ts#L33), which is of type Model<T>.

Reply:
Alternatively if you do `const scholarship = Scholarship.new(data)`, `scholarship.id` will also hold the ID.

Reply:
Ooo that makes sense! Ok I'll add the action to view the scholarship. Thanks!